src/components/settings-menu-button.js

```diff
@@ -0,0 +1,80 @@
+
+/**
+ * @file settings-menu-button.js
+ */
+import videojs from 'video.js';
+import SettingsMenuItem from './settings-menu-item.js';
+const MenuButton = videojs.getComponent('MenuButton');
+const Menu = videojs.getComponent('Menu');
+const Component = videojs.getComponent('Component');
+
+/**
+ * The component for controlling the settings menu
+ *
+ * @param {Player|Object} player
+ * @param {Object=} options
+ * @extends MenuButton
+ * @class SettingsMenuButton
+ */
+class SettingsMenuButton extends MenuButton {
+
+  constructor(player, options) {
+    super(player, options);
+
+    this.el_.setAttribute('aria-label', 'Settings Menu');
+
+    this.on('mouseleave', videojs.bind(this, this.hideChildren));
+  }
+
+  /**
+   * Allow sub components to stack CSS class names
+   *
+   * @return {String} The constructed class name
+   * @method buildCSSClass
+   */
+  buildCSSClass() {
+    // vjs-icon-cog can be removed when the settings menu is integrated in video.js
+    return `vjs-settings-menu vjs-icon-cog ${super.buildCSSClass()}`;
+  }
+
+  /**
+   * Create the settings menu
+   *
+   * @return {Menu} Menu object populated with items
+   * @method createMenu
+   */
+  createMenu() {
+    let menu = new Menu(this.player());
+    let entries = this.options_.entries;
+
+    if (entries) {
+      for (let entry of entries) {
+
+        let settingsMenuItem = new SettingsMenuItem(this.player(), this.options_, entry);
+
+        menu.addChild(settingsMenuItem);
+
+        // Hide children to avoid sub menus stacking on top of each other
+        // or having multiple menus open
+        settingsMenuItem.on('menuitemclicked', videojs.bind(this, this.hideChildren));
```

Comment:
What about here listening to `click` and in `hideChildren` only hide the submenu of the non-selected items? This would mean that we want to have a concept of "selected" settings menu item. And I think it makes sense to have because the selected one is the one that has its sub menus open.